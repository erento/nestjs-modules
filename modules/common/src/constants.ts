export const REQUEST_UNIQUE_ID_BREADCRUMBS_KEY: string = 'nestRequestUniqueIdForBreadcrumbs';
export const REQUEST_UNIQUE_ID_KEY: string = 'nestRequestUniqueId';
