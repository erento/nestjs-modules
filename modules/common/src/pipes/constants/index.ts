export enum ErentoLocale {
    DeDe = 'de-DE',
    DeCh = 'de-CH',
    DeAt = 'de-AT',
    EnGb = 'en-GB',
    EnUs = 'en-US',
}

export const ErentoGermanLocales = [
    ErentoLocale.DeDe,
    ErentoLocale.DeCh,
    ErentoLocale.DeAt,
];

export const languageHosts: Record<ErentoLocale, string> = {
    [ErentoLocale.DeAt]: 'https://www.erento.at',
    [ErentoLocale.DeCh]: 'https://www.erento.ch',
    [ErentoLocale.DeDe]: 'https://www.erento.com',
    [ErentoLocale.EnGb]: 'https://www.erento.co.uk',
    [ErentoLocale.EnUs]: 'https://www.erento.net',
};

export enum CampandaLocale {
    DeDe = 'de_DE',
    DeCh = 'de_CH',
    DeAt = 'de_AT',
    EnGb = 'en_GB',
    EnUs = 'en_US',
    FrFr = 'fr_FR',
    EsEs = 'es_ES',
    ItIt = 'it_IT',
}

export enum ErentoLanguage {
    De = 'de',
    En = 'en',
}

export enum CampandaLanguage {
    De = 'de',
    En = 'en',
    Fr = 'fr',
    Es = 'es',
    It = 'it',
}

export const CAMPANDA_LOCALE_ARRAY: string[] = Object.values(CampandaLocale);
export const CAMPANDA_DEFAULT_LOCALE: string = CampandaLocale.DeDe;

export const ERENTO_LOCALE_ARRAY: string[] = Object.values(ErentoLocale);
export const ERENTO_DEFAULT_LOCALE: string = ErentoLocale.DeDe;
