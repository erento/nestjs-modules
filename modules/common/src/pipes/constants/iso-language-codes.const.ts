export const ISO_LANGUAGE_CODES: string[] = [
    'de-DE',
    'de-CH',
    'de-AT',
    'en-GB',
    'en-US',
    'fr-FR',
    'es-ES',
    'it-IT',
];
