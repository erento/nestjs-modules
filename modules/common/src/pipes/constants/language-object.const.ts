import {LanguageObject} from '../interfaces';

export const GERMAN_LANG_OBJECT: LanguageObject = {
    full: 'de-DE',
    locale: 'de',
    culture: 'DE',
};

export const SWISS_LANG_OBJECT: LanguageObject = {
    full: 'de-CH',
    locale: 'de',
    culture: 'CH',
};

export const AUSTRIAN_LANG_OBJECT: LanguageObject = {
    full: 'de-AT',
    locale: 'de',
    culture: 'AT',
};

export const BRITISH_LANG_OBJECT: LanguageObject = {
    full: 'en-GB',
    locale: 'en',
    culture: 'GB',
};

export const US_LANG_OBJECT: LanguageObject = {
    full: 'en-US',
    locale: 'en',
    culture: 'US',
};

export const LANGUAGES: LanguageObject[] = [
    GERMAN_LANG_OBJECT,
    SWISS_LANG_OBJECT,
    AUSTRIAN_LANG_OBJECT,
    BRITISH_LANG_OBJECT,
    US_LANG_OBJECT,
];
