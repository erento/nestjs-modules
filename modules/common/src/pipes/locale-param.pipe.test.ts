import {CAMPANDA_DEFAULT_LOCALE, CAMPANDA_LOCALE_ARRAY, ERENTO_DEFAULT_LOCALE, ERENTO_LOCALE_ARRAY} from './constants';
import {LocaleParamPipe} from './locale-param.pipe';

describe('LocaleParamPipe', (): void => {
    test.each<[any]>([
        [undefined],
        [null],
        [{}],
        [{locale: 'de-DE'}],
        [{locale: 'de-CH'}],
        [{locale: 'de-AT'}],
        [{locale: 'en-GB'}],
        [{locale: 'en-US'}],
        [{locale: 'de'}],
        [{locale: 'en'}],
        [{locale: 'abc'}],
        [{locale: 'en_GB'}], // fails to find it returns default
        [{locale: 'es-ES'}], // fails to find it returns default
        [{locale: 'es'}], // fails to find it returns default
    ])('should transform erento locale params properly', (input: any): void => {
        const pipe: LocaleParamPipe = new LocaleParamPipe(ERENTO_LOCALE_ARRAY, ERENTO_DEFAULT_LOCALE);
        expect(pipe.transform(input)).toMatchSnapshot();
    });

    test.each<[any]>([
        [undefined],
        [null],
        [{}],
        [{locale: 'de_DE'}],
        [{locale: 'de_CH'}],
        [{locale: 'de_AT'}],
        [{locale: 'en_GB'}],
        [{locale: 'en_US'}],
        [{locale: 'es_ES'}],
        [{locale: 'fr_FR'}],
        [{locale: 'it_IT'}],
        [{locale: 'de'}],
        [{locale: 'en'}],
        [{locale: 'es'}],
        [{locale: 'fr'}],
        [{locale: 'it'}],
        [{locale: 'abc'}],
        [{locale: 'en-GB'}], // fails to find it returns default
    ])('should transform campanda locale params properly', (input: any): void => {
        const pipe: LocaleParamPipe = new LocaleParamPipe(CAMPANDA_LOCALE_ARRAY, CAMPANDA_DEFAULT_LOCALE);
        expect(pipe.transform(input)).toMatchSnapshot();
    });
});
