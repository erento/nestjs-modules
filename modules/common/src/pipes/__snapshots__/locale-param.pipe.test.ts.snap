// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleParamPipe should transform campanda locale params properly 1`] = `
{
  "language": "de",
  "locale": "de_DE",
  "original": "",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 2`] = `
{
  "language": "de",
  "locale": "de_DE",
  "original": "",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 3`] = `
{
  "language": "de",
  "locale": "de_DE",
  "original": "",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 4`] = `
{
  "language": "de",
  "locale": "de_DE",
  "original": "de_DE",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 5`] = `
{
  "language": "de",
  "locale": "de_CH",
  "original": "de_CH",
  "territory": "CH",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 6`] = `
{
  "language": "de",
  "locale": "de_AT",
  "original": "de_AT",
  "territory": "AT",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 7`] = `
{
  "language": "en",
  "locale": "en_GB",
  "original": "en_GB",
  "territory": "GB",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 8`] = `
{
  "language": "en",
  "locale": "en_US",
  "original": "en_US",
  "territory": "US",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 9`] = `
{
  "language": "es",
  "locale": "es_ES",
  "original": "es_ES",
  "territory": "ES",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 10`] = `
{
  "language": "fr",
  "locale": "fr_FR",
  "original": "fr_FR",
  "territory": "FR",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 11`] = `
{
  "language": "it",
  "locale": "it_IT",
  "original": "it_IT",
  "territory": "IT",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 12`] = `
{
  "language": "de",
  "locale": "de_DE",
  "original": "de",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 13`] = `
{
  "language": "en",
  "locale": "en_GB",
  "original": "en",
  "territory": "GB",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 14`] = `
{
  "language": "es",
  "locale": "es_ES",
  "original": "es",
  "territory": "ES",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 15`] = `
{
  "language": "fr",
  "locale": "fr_FR",
  "original": "fr",
  "territory": "FR",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 16`] = `
{
  "language": "it",
  "locale": "it_IT",
  "original": "it",
  "territory": "IT",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 17`] = `
{
  "language": "de",
  "locale": "de_DE",
  "original": "abc",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform campanda locale params properly 18`] = `
{
  "language": "de",
  "locale": "de_DE",
  "original": "en-GB",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 1`] = `
{
  "language": "de",
  "locale": "de-DE",
  "original": "",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 2`] = `
{
  "language": "de",
  "locale": "de-DE",
  "original": "",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 3`] = `
{
  "language": "de",
  "locale": "de-DE",
  "original": "",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 4`] = `
{
  "language": "de",
  "locale": "de-DE",
  "original": "de-DE",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 5`] = `
{
  "language": "de",
  "locale": "de-CH",
  "original": "de-CH",
  "territory": "CH",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 6`] = `
{
  "language": "de",
  "locale": "de-AT",
  "original": "de-AT",
  "territory": "AT",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 7`] = `
{
  "language": "en",
  "locale": "en-GB",
  "original": "en-GB",
  "territory": "GB",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 8`] = `
{
  "language": "en",
  "locale": "en-US",
  "original": "en-US",
  "territory": "US",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 9`] = `
{
  "language": "de",
  "locale": "de-DE",
  "original": "de",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 10`] = `
{
  "language": "en",
  "locale": "en-GB",
  "original": "en",
  "territory": "GB",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 11`] = `
{
  "language": "de",
  "locale": "de-DE",
  "original": "abc",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 12`] = `
{
  "language": "de",
  "locale": "de-DE",
  "original": "en_GB",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 13`] = `
{
  "language": "de",
  "locale": "de-DE",
  "original": "es-ES",
  "territory": "DE",
}
`;

exports[`LocaleParamPipe should transform erento locale params properly 14`] = `
{
  "language": "de",
  "locale": "de-DE",
  "original": "es",
  "territory": "DE",
}
`;
